src/gl_fns.rs

```diff
@@ -295,7 +295,8 @@ impl Gl for GlFns {
 
     // https://www.khronos.org/registry/OpenGL-Refpages/es2.0/xhtml/glGetUniform.xml
     fn get_uniform_iv(&self, program: GLuint, location: GLint) -> Vec<GLint> {
-        let len = get_uniform_iv_vector_length(&location);
+        let (_, uniform_type, _) = self.get_active_uniform(program, 0);
```

Comment:
Just looking over `get_active_uniform` now, it appears to me that we can have a lightweight version for internal use to only get that uniform type, avoiding multiple GL calls and heap allocations.

Reply:
Sure.
If we implement it alongside these functions, it will be exposed outside of gleam, which is something we probably don't want, right? Instead, we could define a private trait with the internal function and implement it for `GlFns` and `GlesFns`. Sounds alright?

Reply:
Oh, we can just `impl GlFns`, then we can add private methods too. That's simpler.

Reply:
Updated the PR.